src/android/io/sqlc/SQLitePlugin.java

```diff
@@ -337,6 +337,11 @@ private static String getDBConnectionName(String dbname, JSONObject options) {
 
             this.q = new LinkedBlockingQueue<DBQuery>();
             this.openCbc = cbc;
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                Log.v(SQLitePlugin.class.getSimpleName(), "Applying hotfix for Android 11+");
+                this.oldImpl = true;
```

Comment:
What does this do? Is it equivalent to use the `androidDatabaseProvider: 'system'` setting? There are some glitches when using that setting: https://github.com/storesafe/cordova-sqlite-storage#android-database-provider

Reply:
Yes, it is, but it will only be enable for Android 11. Regarding the database lock issue, it seems to no longer be an issue in the most recent Android versions. On the other hand, the data corruption issue appears to be caused by the use of multiple plugins to access the same database.

Reply:
The app always uses the same plugin, so it should not be a problem.